### DISCUSSION TITLE
Rate Limit Exception

### DISCUSSION THREAD

**Discussion by AkikoOrenji** [censys/censys-python#56] [Q&A] [answered]
I notice we have a rate limit on our account:  "_api 1.0 actions/second (300.0 per 5 minute interval)_"

Is this taken into account as part of this python library (is there a built in back off) and what happens if we go over this limit  ?  e.g. are blank records returned or is an error message is returned somewhere etc 

My usage approx usage is 

```
import censys.ipv4
ip  = "1.1.1.1"
c = censys.ipv4.CensysIPv4(api_id="blanked", api_secret="blanked")
c.view(ip)
```

Thanks

**Comment by thehappydinoa** [comment, marked as answer]
Hi there,

Yes so we have backoff installed with the library, but if the rate limit is reached the API will raise a `CensysRateLimitExceededException`. Please let me know if you have any more questions.
